www/class/config-generate/generate.class.php

```diff
@@ -113,17 +125,35 @@ private function generateIndexData($localhost = 0)
                 $meta_services = MetaService::getInstance($this->dependencyInjector)->getMetaServices();
                 $host_id = MetaHost::getInstance($this->dependencyInjector)->getHostIdByHostName('_Module_Meta');
                 foreach ($meta_services as $meta_id => $meta_service) {
-                    $stmt->bindValue(':host_name', '_Module_Meta', PDO::PARAM_STR);
-                    $stmt->bindValue(':service_description', 'meta_' . $meta_id, PDO::PARAM_STR);
-                    $stmt->bindParam(':host_id', $host_id, PDO::PARAM_INT);
-                    $stmt->bindParam(':service_id', $meta_service['service_id'], PDO::PARAM_INT);
-                    $stmt->execute();
+                    $bulkValues[] = '(' . 
+                        $this->backend_instance->db_cs->quote($host_id, PDO::PARAM_INT) . ',' . 
+                        $this->backend_instance->db_cs->quote($meta_service['service_id'], PDO::PARAM_INT) . ',' .
+                        $this->backend_instance->db_cs->quote('_Module_Meta', PDO::PARAM_STR) . ',' . 
+                        $this->backend_instance->db_cs->quote('meta_' . $meta_id, PDO::PARAM_STR) . ')';
+                    $bulkCount++;
+                    if ($bulkCount == $bulkLimit) {
+                        $this->backend_instance->db_cs->exec(
+                            $query .
+                            implode(',', $bulkValues) .
+                            " ON DUPLICATE KEY UPDATE 
+                            host_name=VALUES(host_name), service_description=VALUES(service_description)"
+                        );
+                        $bulkCount = 0;
+                        $bulkValues = [];
+                    }
                 }
             }
 
-            $this->backend_instance->db_cs->commit();
```

Comment:
to stay ISO, commit and rollback are required

Reply:
we don't need transaction here, it locks table index_data too much time, and it's not necessary